### DISCUSSION TITLE
Count and order for inner foreign tables?

### DISCUSSION THREAD

**Discussion by fayazara** [supabase/supabase#4549] [Questions] [answered]
I want to get the count and change the order of a foriegn table? How do I do this?

basically something like this.

```await this.$supabase.from("forms").select(`name,submissions({count: exact}, {created_at: desc})`);```

**Comment by steve-chavez** [comment, marked as answer]
For the foreign table count, you can do:

```js
supabase.
from("forms").
select(`name,submissions(count)`)
```

**Reply by arcavid** [reply]
How to order by that count?

**Reply by laubonghaudoi** [reply]
it seems like this is not working as of Aug, 2022. Did Supabase change?

**Reply by steve-chavez** [reply]
@laubonghaudoi What error message did you get?

**Reply by laubonghaudoi** [reply]
@steve-chavez Not an error. I just figured that it is working, but the count is returned within a single element-array, like :

```javascript
{
    name: "foo",
    submissions: [
        {
            count: 1
        }  
    ]
}
```
Why is it returned this way? Can I make it like this
```javascript
{
    name: "foo",
    submissions: {
        count: 1
    }
}
```

**Reply by steve-chavez** [reply]
> Why is it returned this way? 

@laubonghaudoi Doing `table(count)` is a bit of a hack right now, it will only respond with an array. Support for aggregates will later solve this.

**Reply by laubonghaudoi** [reply]
I see, thanks for clarifying! Is there an issue or a thread to track this?

**Reply by steve-chavez** [reply]
Yes, see https://github.com/supabase/postgrest-js/issues/206

**Reply by steve-chavez** [reply]
> How to order by that count?

It's possible with a computed column as mentioned on https://github.com/supabase/supabase/discussions/7875

**Comment by KevM** [comment]
The [query docs](https://supabase.com/docs/reference/javascript/select#query-foreign-tables) do not cover this scenario. This Issue was the only place I found an answer.
